internal/cache/util/util_test.go

```diff
@@ -233,3 +234,40 @@ func (ut *utilTest) Test_IsCacheHandleValid_False() {
 		ExpectFalse(IsCacheHandleInvalid(errors.New(errMsg)))
 	}
 }
+
+func TestCreateCacheDirectoryIfNotPresentAtShouldNotReturnAnyErrorWhenDirectoryExists(t *testing.T) {
+	base := path.Join("./", string(testutil.GenerateRandomBytes(4)))
+	dirPath := path.Join(base, "/", "path/cachedir")
+	dirCreationErr := os.MkdirAll(dirPath, 0700)
+	defer os.RemoveAll(base)
+	AssertEq(nil, dirCreationErr)
+
+	err := CreateCacheDirectoryIfNotPresentAt(dirPath)
+
+	AssertEq(nil, err)
+}
+
+func TestCreateCacheDirectoryIfNotPresentAtShouldNotReturnAnyErrorWhenDirectoryCanBeCreated(t *testing.T) {
+	base := path.Join("./", string(testutil.GenerateRandomBytes(4)))
+	dirPath := path.Join(base, "/", "path/cachedir")
+	defer os.RemoveAll(base)
+
+	err := CreateCacheDirectoryIfNotPresentAt(dirPath)
+
+	AssertEq(nil, err)
+	fileInfo, err := os.Stat(dirPath)
+	AssertEq(nil, err)
+	AssertEq(0755, fileInfo.Mode().Perm())
```

Comment:
nit: should we reuse the FileDirPerm constant here ?

Reply:
delibaretly added hard coded value so tests fails if FileDirPerm value is changed anytime, to make sure we explicitly know that impact of changing is also on this functionality

Reply:
have slight different view here - the functionality of the function is creating directory with permission stated by FileDirPerm and we are unit testing the functionality.

Reply:
the functionality of this function is to create a writable directory, we used FileDirPerm only for reusability purpose, imo we should test the functionality of writable directory and not link it with FileDirPerm value, does it make sense, what do you think ?